### ISSUE TITLE
_Action with form example_ does not work

### ISSUE THREAD

**Issue by raptium** [unfoldadmin/django-unfold#526]
I am using django-unfold 0.29.1 with django 5.0.6

The example snippet under README [Action with form example](https://github.com/unfoldadmin/django-unfold?tab=readme-ov-file#action-with-form-example) does not work.

```python
@register(User)
class UserAdmin(ModelAdmin):
    actions_detail = ["change_detail_action_block"]
    form = SomeForm(request.POST or None)  # <<< WHERE DOES request COME FROM?

    @action(description=_("Detail"))
    def change_detail_action_block(self, request: HttpRequest, object_id: int) -> str:
        user = User.objects.get(pk=object_id)

        if request.method == "POST" and form.is_valid():
            # Do something with form data
            form.cleaned_data["note"]

            return redirect(
                reverse_lazy("admin:users_user_change", args=[object_id])
            )

        return render_to_string("some/template.html", {
            "form": form,
        })
```

The server does not start since there is no `request` in context, and I guess that line should go in `change_detail_action_block`.

After moving the line, I see another error when I click the action button.

```
AttributeError at /admin/example/post/1/change_detail_action_block/
'SafeString' object has no attribute 'has_header'
Request Method:	GET
Request URL:	http://127.0.0.1:8000/admin/example/post/1/change_detail_action_block/
Django Version:	5.0.6
Exception Type:	AttributeError
Exception Value:	
'SafeString' object has no attribute 'has_header'
Exception Location:	/home/guanhao/pets/django-apps/.venv/lib/python3.11/site-packages/django/utils/cache.py, line 284, in patch_response_headers
Raised during:	unfold.decorators.inner
Python Executable:	/home/guanhao/pets/django-apps/.venv/bin/python
Python Version:	3.11.9
Python Path:	
['/home/guanhao/pets/django-apps/reproduce',
 '/home/guanhao/.pyenv/versions/3.11.9/lib/python311.zip',
 '/home/guanhao/.pyenv/versions/3.11.9/lib/python3.11',
 '/home/guanhao/.pyenv/versions/3.11.9/lib/python3.11/lib-dynload',
 '/home/guanhao/pets/django-apps/.venv/lib/python3.11/site-packages']
Server time:	Sat, 29 Jun 2024 01:13:24 +0000
```

It seems the action function should return a response instead of the rendered string. So I changed it to

```python
        return render(request, "some/template.html", {
            "form": form,
        })
```

When I view the form example this time, no error is displayed, and I can see a form with one field. However, I'm unsure if this matches the intended example. Additionally, the example snippet lacks some import statements, which makes it less straightforward to use compared to other examples that work with simple copy and paste.

**Comment by lukasvinclav** [comment]
Fixed #527

**Comment by yswtrue** [comment]
it still get same error after follow the document
